server/src/main/java/io/crate/expression/tablefunctions/GenerateSubscripts.java

```diff
@@ -0,0 +1,184 @@
+    private static final RowType RETURN_TYPE = new RowType(List.of(INTEGER), List.of(NAME.name()));
+
+
+    public static void register(TableFunctionModule module) {
+        module.register(
+            Signature.table(
+                NAME,
+                parseTypeSignature("array(E)"),
+                DataTypes.INTEGER.getTypeSignature(),
+                DataTypes.INTEGER.getTypeSignature()
+            ).withTypeVariableConstraints(typeVariable("E")),
+            GenerateSubscripts::new
+        );
+        module.register(
+            Signature.table(
+                NAME,
+                parseTypeSignature("array(E)"),
+                DataTypes.INTEGER.getTypeSignature(),
+                DataTypes.BOOLEAN.getTypeSignature(),
+                DataTypes.INTEGER.getTypeSignature()
+            ).withTypeVariableConstraints(typeVariable("E")),
+            GenerateSubscripts::new
+        );
+    }
+
+    private final Signature signature;
+    private final Signature boundSignature;
+
+    private GenerateSubscripts(Signature signature, Signature boundSignature) {
+        this.signature = signature;
+        this.boundSignature = boundSignature;
+    }
+
+    private static int getNumRows(List<?> array, int depthLevel) {
+        if (depthLevel <= 0) {
+            throw new IllegalArgumentException("target level must be greater than zero");
+        }
+        List<?> targetArray = array;
+        for (int level = 2; level <= depthLevel; level++) {
+            if (targetArray == null || targetArray.isEmpty()) {
+                return 0;
+            }
+            int size = -1;
+            List<?> firstNonNullElement = null;
+            for (int i = 0; i < targetArray.size(); i++) {
+                Object oi = targetArray.get(i);
+                if (oi == null) {
+                    // null is a valid value within an array
+                    continue;
+                }
+                if (!(oi instanceof List)) {
+                    return 0;
+                }
+                List<?> element = (List<?>) oi;
+                if (size == -1) {
+                    size = element.size();
+                } else {
+                    if (size != element.size()) {
+                        throw new IllegalArgumentException(String.format(
+                            Locale.ENGLISH,
+                            "nested arrays must have the same dimension within a level, offending level %d, position %d",
+                            level, i + 1));
+                    }
+                }
+                if (firstNonNullElement == null) {
+                    firstNonNullElement = element;
+                }
+            }
+            targetArray = firstNonNullElement;
+        }
+        return targetArray != null ? targetArray.size() : 0;
+    }
+
+    @SafeVarargs
+    @Override
+    public final Bucket evaluate(TransactionContext txnCtx, Input<T>... args) {
```

Comment:
Why the change to `Bucket`? This can still be an Iterable, the additional size info of `Bucket` is not required or used.